db/post_migrate/20220307083603_optimize_null_indexes.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class OptimizeNullIndexes < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction!
+
+  def up
+    update_index :conversations, 'index_conversations_on_uri', :uri, unique: true, where: 'uri IS NOT NULL'
+    update_index :statuses, 'index_statuses_on_in_reply_to_account_id', :in_reply_to_account_id, where: 'in_reply_to_account_id IS NOT NULL'
+    update_index :statuses, 'index_statuses_on_in_reply_to_id', :in_reply_to_id, where: 'in_reply_to_id IS NOT NULL'
+    update_index :media_attachments, 'index_media_attachments_on_scheduled_status_id', :scheduled_status_id, where: 'scheduled_status_id IS NOT NULL'
```

Comment:
I don't think we should do that, pretty sure the index is used for the `unattached` scope.

Reply:
In practice, unattached scope uses `index_media_attachments_on_status_id`, and simply filters over `scheduled_status_id`

Reply:
Hm, it makes sense, `index_media_attachments_on_status_id` allows finding everything that is either to-be-deleted or scheduled. Though performance issues would arise again if the number of scheduled toots increases.

Reply:
So I think it's fine to change it, because it's indeed wasteful as it would be `NULL` for the overwhelming majority of statuses, but it might make sense to have a partial index where both `status_id` and `scheduled_status_id` values are `NULL`.